sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -342,18 +342,33 @@ private static string GetCanonicalName(string account, string fileSystemName, st
         /// Ensure the <see cref="DataLakeSasBuilder"/>'s properties are in a
         /// consistent state.
         /// </summary>
-        private void EnsureState()
+        internal void EnsureState()
         {
-            if (ExpiresOn == default)
+            // Identifier is not present
+            if (string.IsNullOrEmpty(Identifier))
             {
-                throw Errors.SasMissingData(nameof(ExpiresOn));
+                if (string.IsNullOrEmpty(Permissions))
+                {
+                    throw Errors.SasMissingData(nameof(Permissions));
+                }
+
+                if (ExpiresOn == default)
+                {
+                    throw Errors.SasMissingData(nameof(ExpiresOn));
+                }
             }
-            if (string.IsNullOrEmpty(Permissions))
+
+            // Identifier is present
+            else
             {
-                throw Errors.SasMissingData(nameof(Permissions));
+                if (!string.IsNullOrEmpty(Permissions))
+                {
+                    throw Errors.SasDataNotAllowed(nameof(Identifier), nameof(Permissions));
+                }
             }
```

Comment:
For signed permission and expirty on SAS "This field must be omitted if it has been specified in an associated stored access policy." so we shouldnt check this, also add documentation specifying this

Reply:
fixed.